src/extension.js

```diff
@@ -0,0 +1,49 @@
+const path = require('path')
+
+const {workspace} = require('vscode')
+const {LanguageClient} = require('vscode-languageclient/node')
+
+/**
+ * @param {import('vscode').ExtensionContext} context
+ */
+function activate(context) {
+  const command = require.resolve('.bin/remark-language-server')
+  const args = ['--stdio']
+
+  /**
+   * @type {import('vscode-languageclient/node').ServerOptions}
+   */
+  const serverOptions = {
+    run: {command, args},
+    debug: {
+      command: 'node',
+      args: ['--inspect=6009', command, ...args]
+    }
+  }
+
+  /**
+   * @type {import('vscode-languageclient').LanguageClientOptions}
+   */
+  const clientOptions = {
+    documentSelector: [{scheme: 'file', language: 'markdown'}],
+    synchronize: {
+      fileEvents: [
+        workspace.createFileSystemWatcher(
+          '**/.remark{ignore,rc,rc.cjs,rc.js,rc.json,rc.mjs,rc.yaml,rc.yml}'
```

Comment:
does unified surface the file types it supports as a constant? :thinking: 
If not would it make sense to have a constant so downstream packages can be kept in sync?

Reply:
I don’t think it does. Even if it did, it uses ESM, which can’t be consumed by Visual Studio Code extensions.

Also it’s not a big deal. The extensions are unlikely to change and even if they do, the impact is low.